.github/workflows/crdb-api-client-npm-publish.yml

```diff
@@ -0,0 +1,110 @@
+# This workflow establishes two-step process of building and publishing `crdb-api-client` NPM package
+# when the produced outputs are changed comparing to last published version.
+# 1st step (first CI run) checks that built files contain changes comparing to previous version and
+# if NPM package version incremented:
+# - if yes, it publishes to NPM registry
+# - if no, new PR created (with incremented version in package.json file) which should be merged and
+# then second step is executed (runs workflow again) and publishes package.
+name: Publish crdb-api-client package to NPM registry
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - master
+
+jobs:
+  publish_crdb_api_client:
+    if: github.repository == 'cockroachdb/cockroach'
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v3
+      with:
+        submodules: true
+
+    - name: Bazel Cache
+      uses: actions/cache@v3
+      with:
+        path: ~/.cache/bazel
+        key: ${{ runner.os }}-bazel-cache
+
+    - uses: pnpm/action-setup@v2
+      with:
+        version: "8.6.10"
+
+    - name: Setup NodeJS
+      uses: actions/setup-node@v3
+      with:
+        node-version: 16
+        registry-url: 'https://registry.npmjs.org'
+        always-auth: true
+        cache: 'pnpm'
+        cache-dependency-path: "${{ github.workspace }}/pkg/ui/pnpm-lock.yaml"
+      env:
+        NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
+
+    - name: Build crdb-api-client package
+      run: |
+        echo "build --config=dev" >> .bazelrc.user
+        echo "build --config nolintonbuild" >> .bazelrc.user
+        echo "test --test_tmpdir=/tmp/cockroach" >> .bazelrc.user
+        ./dev doctor --interactive=false --no-cache
+        ./dev ui crdb-api-client
```

Comment:
Is there a way to run the same thin without using `dev`? We tend to use `dev` for local runs only and prefer using explicit `bazel` commands for CI. This will also eliminate the need to change the `.bazelrc.user` file.